server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java

```diff
@@ -125,8 +125,15 @@ public static void createField(DocumentParserContext context, BytesRef tsid) {
         ByteUtils.writeLongBE(timestamp, suffix, 8);   // Big Ending shrinks the inverted index by ~37%
```

Comment:
I would love to understand why BE yields such savings. Also, why do we write the tsid hash in LE?

This is a question for my understanding, not a comment on the code. Thanks

Reply:
It's like 30%. It's because in big endian the high bits change very little so the search index ends up with longer common prefixes. We use little endian for writing the hash because either direction is fine and we *expect* little endian the be faster because we *expect* it is the processor's native endianness.